### PR TITLE
enhance search online by National ID from the patient search widget.

### PR DESCRIPTION
**Pull request by slubwama**
https://trello.com/c/NFRfGotw/746-fingerprint-eliminate-viewing-of-previously-searched-patients-on-the-search-page

https://trello.com/c/cAt3Zz9M/747-make-online-search-by-nin-be-done-at-the-patient-search-widget

https://github.com/METS-Programme/openmrs-module-ugandaemr-fingerprint/issues/21

https://github.com/METS-Programme/openmrs-module-ugandaemr-fingerprint/issues/22

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/ugandaemrfingerprint/core/FingerPrintConstant.java [hunk 1]
```diff
@@ -28,14 +28,23 @@
...
+    public static final String MOST_RECENT_ENCOUNTER = "{patient(patients: [\"%s\"]) {uuid birthdate gender dead patientFacility {uuid name} names {familyName middleName givenName voided} mostRecentEncounter {obs {uuid concept encounterDate encounterType valueCoded valueText valueNumeric valueDatetime valueDrug valueBoolean valueComplex voided}}}}";
+    public static final String SUMMARY_ENCOUNTER = "{patient(patients: [\"%s\"]) {uuid birthdate gender dead patientFacility {uuid name} names {familyName middleName givenName voided} summaryPage {obs {uuid concept encounterDate encounterType valueCoded valueText valueNumeric valueDatetime valueDrug valueBoolean valueComplex voided}}}}";
     public static final String PATIENT_ONLINE_ID = "identifiers";
     public static final String PATIENT_NOT_FOUND = "Patient Not Found";
     public static final String CONNECTION_TEST_IP = "google.com/";
     public static final String DEVICE_SOCKET_IP = "ugandaemrfingerprint.socketIPAddress";
-    public static final String DEVICE_SOCKET_IP_PLACE_HOLDER = "/complete";
+    public static final String DEVICE_SOCKET_IP_PLACE_HOLDER = "http://localhost:8084";
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrfingerprint/core/FingerPrintConstant.java:40]: @slubwama Why the hardcoded port?

File: api/src/main/java/org/openmrs/module/ugandaemrfingerprint/core/FingerPrintConstant.java [hunk 2]
```diff
@@ -28,14 +28,23 @@
...
+    public static final String PATIENT_DETAILS = "{patient(patients: [\"%s\"]) {uuid birthdate gender dead patientFacility {uuid name} names {familyName middleName givenName voided}}}";
+    public static final String MOST_RECENT_ENCOUNTER = "{patient(patients: [\"%s\"]) {uuid birthdate gender dead patientFacility {uuid name} names {familyName middleName givenName voided} mostRecentEncounter {obs {uuid concept encounterDate encounterType valueCoded valueText valueNumeric valueDatetime valueDrug valueBoolean valueComplex voided}}}}";
+    public static final String SUMMARY_ENCOUNTER = "{patient(patients: [\"%s\"]) {uuid birthdate gender dead patientFacility {uuid name} names {familyName middleName givenName voided} summaryPage {obs {uuid concept encounterDate encounterType valueCoded valueText valueNumeric valueDatetime valueDrug valueBoolean valueComplex voided}}}}";
     public static final String PATIENT_ONLINE_ID = "identifiers";
     public static final String PATIENT_NOT_FOUND = "Patient Not Found";
     public static final String CONNECTION_TEST_IP = "google.com/";
     public static final String DEVICE_SOCKET_IP = "ugandaemrfingerprint.socketIPAddress";
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrfingerprint/core/FingerPrintConstant.java:39]: Consider adding a prefix to Global properties to make them stand out 

```suggestion
    public static final String GP_DEVICE_SOCKET_IP = "ugandaemrfingerprint.socketIPAddress";
```

File: api/src/main/java/org/openmrs/module/ugandaemrfingerprint/core/FingerPrintConstant.java (hunk 1 above)
Comment (slubwama) [reply api/src/main/java/org/openmrs/module/ugandaemrfingerprint/core/FingerPrintConstant.java:40]: Its a place Holder not the actual the Socket IP address. Check DEVICE_SOCKET_IP

File: api/src/main/java/org/openmrs/module/ugandaemrfingerprint/core/FingerPrintConstant.java (hunk 2 above)
Comment (slubwama) [reply api/src/main/java/org/openmrs/module/ugandaemrfingerprint/core/FingerPrintConstant.java:39]: Noted

**Pull request closed without merging**